transfer_results_module/transfer_results.py

```diff
@@ -146,19 +160,21 @@ def run_transfer_results():
     # first check if we have results stored locally, that have not been inserted in MySQL
     results_filename = 'results.json'
     if os.path.isfile(results_filename):
-        logging.warning("there are results in %s that we didnt't insert. do it now!" % results_filename)
-        print("there are results in %s, that we didnt't insert. do it now!" % results_filename)
         # start to import the old results first
         with open(results_filename) as results_file:
             results = json.load(results_file)
-            transfer_results(results_filename, results)
+            results_txt_filename = results_to_txt(results)
+            logging.warning("there are results in %s that we didnt't insert. do it now!" % results_filename)
+            save_results_mysql(results_txt_filename)
```

Comment:
I think there might be some sneaky issues that could come up with this that are worth outlining and trying to account for. Namely, what happens if something goes wrong within `save_results_mysql`? An intermittent connection issue, timeout, or something else could cause this worker to be unable to save the results properly. Then, the workflow would be:

* Put results in text file
* Try to save results to MySQL
  * Saving to MySQL fails
* Delete the results in the file

which means we just lost everything. 

To make it more resilient, it'd be better to have `save_results_mysql` raise some sort of exception so that you can catch it. If things go "as planned" then proceed to delete the results from the file. Otherwise, you'll want to bail and not delete the result file until you can properly save them.

Reply:
Hey,
when starting the script, the first thing that we do is to download the results from firebase and put everything into a `results.json` file. This file is deleted, when everything went fine at the end of the script. In case an error happens somewhere in between, the `results.json` file is not removed. 

When the script is started, it will check if there is a results.json file and if so, start to import the results that are stored there.

I'm not sure, if this already addressed the issue. If we reach a timeout, will the script not throw an error and stop at the position, where the error happened?

Reply:
Furthermore, we would not loose results, untill there are not deleted in firebase. When save_to_mysql fails then we will not delete the entries in firebase. The next time the script is started it will therefore also import these results again.

Reply:
But, I'm thinking on "what else could go wrong".

Our main problem would be if any results are lost. This means that results are not imported to the mysql, but deleted in firebase. "Old" results are lost after the "results.json" file is deleted. We have to make sure, that the file is not deleted if anything goes wrong.

Reply:
Sorry Benni, on a second look, I think you might have those bases covered. There are still some parts that make me a little uneasy, though, so I'd invite any comments or suggestions from @ericboucher if he has any